app/cni/pkg/install/main.go

```diff
@@ -25,17 +26,26 @@ const (
 	secondaryBinDir    = "/host/secondary-bin-dir"
 	serviceAccountPath = "/var/run/secrets/kubernetes.io/serviceaccount"
 	readyFilePath      = "/tmp/ready"
+	defaultLogName     = "install-cni"
 )
 
 var (
-	log = core.NewLoggerTo(os.Stderr, 2).WithName("install-cni")
+	log             = CreateNewLogger(defaultLogName, kuma_log.DebugLevel)
+	cleanupFinished = false
+	cleanupMutex    = sync.Mutex{}
 )
 
 func removeBinFiles() error {
 	return os.Remove("/host/opt/cni/bin/kuma-cni")
 }
 
 func cleanup(ic *InstallerConfig) {
+	cleanupMutex.Lock()
+	defer cleanupMutex.Unlock()
+	if cleanupFinished {
+		return
+	}
```

Comment:
Does that mean that cleanup is called multiple times?

Reply:
we do a `defer` and "signal exit" here - https://github.com/kumahq/kuma/blob/ad73606e83780877ace03c0b3cddd2cb0029caa4/app/cni/pkg/install/main.go#L256 - I'm not sure `defer` will always be called if a process is killed via signal, so I saw in the logs that it was called multiple times. If there are guarantees on `defer` being called I can remove the signal cleanup.